addons/account/wizard/account_payment_register.py

```diff
@@ -766,12 +772,14 @@ def _create_payment_vals_from_batch(self, batch_result):
             'company_id': self.company_id.id,
             'currency_id': batch_values['source_currency_id'],
             'partner_id': batch_values['partner_id'],
-            'partner_bank_id': partner_bank_id,
             'payment_method_line_id': payment_method_line.id,
             'destination_account_id': batch_result['lines'][0].account_id.id,
             'write_off_line_vals': [],
         }
 
+        if partner_bank_id:
+            payment_vals['partner_bank_id'] = partner_bank_id
```

Comment:
I wouldn't mind a comment here, otherwise people seeing that code will tend to move it back where it was and reintroduce the behavior we're trying to fix :)

Reply:
It might be my fault 🤣 
https://github.com/odoo/odoo/pull/158133#discussion_r1533306721

Reply:
:D haha well, yeah, I think it's useful because the reason isn't obvious